debugger/src/debug_adapter/repl_commands.rs

```diff
@@ -0,0 +1,774 @@
+                            "\n{} [{} @ {}]: {} ",
+                            variable.name,
+                            variable.type_name,
+                            variable.memory_location,
+                            variable.get_value(variable_cache)
+                        ));
+                    }
+                }
+            } else {
+                return Err(DebuggerError::ReplError(format!(
+                    "No variable named {:?} found for frame: {:?}.",
+                    variable_name, stack_frame.function_name
+                )));
+            }
+        } else {
+            return Err(DebuggerError::ReplError(format!(
+                "No variables available for frame: {:?}.",
+                stack_frame.function_name
+            )));
+        }
+    } else {
+        return Err(DebuggerError::ReplError("No frame selected.".to_string()));
+    }
+    Ok(DebugSessionStatus::Continue)
+}
+
+/// Read memory at the specified address (hex), using the [`GdbNuf`] specifiers to determine size and format.
+fn memory_read(
+    address: u64,
+    gdb_nuf: GdbNuf,
+    target_core: &mut CoreHandle,
+    response_body: &mut EvaluateResponseBody,
+) -> Result<DebugSessionStatus, DebuggerError> {
+    if gdb_nuf.format_specifier == GdbFormat::Instruction {
+        let assembly_lines: Vec<DisassembledInstruction> = disassemble_target_memory(
+            target_core,
+            0_i64,
+            0_i64,
+            address,
+            gdb_nuf.unit_count as i64,
+        )?;
+        if assembly_lines.is_empty() {
+            return Err(DebuggerError::ReplError(format!(
+                "Cannot disassemble memory at address {address:#010x}"
+            )));
+        } else {
+            let mut formatted_output = "".to_string();
+            for assembly_line in &assembly_lines {
+                formatted_output.push_str(&assembly_line.to_string());
+            }
+            response_body.result = formatted_output;
+        }
+    } else {
+        let mut memory_result = vec![0u8; gdb_nuf.get_size()];
+        match target_core.core.read_8(address, &mut memory_result) {
+            Ok(()) => {
+                let formatted_output = GdbNufMemoryResult {
+                    nuf: &gdb_nuf,
+                    memory: &memory_result,
+                }
+                .to_string();
+                response_body.result = formatted_output;
+            }
+            Err(err) => {
+                return Err(DebuggerError::ReplError(format!(
+                    "Cannot read memory at address {address:#010x}: {err:?}"
+                )))
+            }
+        }
+    }
+    Ok(DebugSessionStatus::Continue)
+}
+
+/// Get a list of command matches, based on the given command piece.
+/// The `command_piece` is a valid [`ReplCommand`], which can be either a command or a sub_command.
+fn find_commands<'a>(
+    repl_commands: Vec<&'a ReplCommand<H>>,
```

Comment:
This could be a slice, something like `&'b [&'a ReplCommand<H>]`, then there's no need to allocate a whole new `Vec`.

Reply:
Good catch. Thanks :)